admin/load.php

```diff
@@ -131,8 +131,8 @@ function perflab_render_pointer( $pointer_id = 'perflab-admin-pointer', $args =
 	<script id="<?php echo esc_attr( $pointer_id ); ?>" type="text/javascript">
 		jQuery( function() {
 			// Pointer Options.
-			var options = {
-				content: '<h3><?php echo esc_js( $args['heading'] ); ?></h3><p><?php echo wp_kses( $args['content'], $wp_kses_options ); ?></p>',
+			const options = {
+				content: <?php echo wp_json_encode( '<h3>' . esc_html( $args['heading'] ) . '</h3><p>' . wp_kses( $args['content'], $wp_kses_options ) . '</p>' ); ?>,
```

Comment:
Why is `wp_json_encode()` better here? It isn't really JSON, so this feels a bit weird to me semantically.

Reply:
Well, a few reasons:

1. It avoids the author having to keep track of the open and closing quotation marks, and remember whether they are single or double.
2. It consolidates the echoes into one rather than splitting it into two.

This also fixes a couple issues:

1. There is no escaping of the heading being done, so it's adding an `esc_html()`. If using `esc_js()` then it should probably should have been `esc_js( esc_html( $args['heading'] ) )`
2. Similarly, the `wp_kses( $args['content'], $wp_kses_options )` should have also been wrapped in `esc_js()` here in case the `content` contains some single quote mark or backslashes or some other character needing escaping.

By using `wp_json_encode()`, the code is made more robust against bugs being introduced.